internal/checkout/source_provider.go

```diff
@@ -201,6 +218,43 @@ func (cfg *Config) validate() error {
 	return nil
 }
 
+func pullRequestInfo(eventContext map[string]interface{}) (*PullRequest, error) {
+	pr, ok := getMapFromMap(eventContext, "pullRequest")
+	if !ok {
+		return nil, nil
+	}
+
+	if ok, _ := getBoolFromMap(pr, "MergeCommitSHASupported"); ok {
```

Comment:
nit: `getBoolFromMap` and `getStringFromMap` could accept a path argument (e.g. "pullRequest.MergeCommitSHASupported") and put `getMapFromMap` there to reduce this function.

Reply:
I would have to change the usage of this function in all other places, it's not just used within my implementation, I would avoid doing that since it's not within the scope of this PR.